tsdb/series_file.go

```diff
@@ -35,17 +37,39 @@ type SeriesFile struct {
 	path       string
 	partitions []*SeriesPartition
 
+	maxCompactionConcurrency int
+
 	refs sync.RWMutex // RWMutex to track references to the SeriesFile that are in use.
 
 	Logger *zap.Logger
 }
 
 // NewSeriesFile returns a new instance of SeriesFile.
 func NewSeriesFile(path string) *SeriesFile {
+	maxCompactionConcurrency := runtime.GOMAXPROCS(0)
+	if maxCompactionConcurrency < 1 {
+		maxCompactionConcurrency = 1
+	}
+	if maxCompactionConcurrency > SeriesFilePartitionN {
```

Comment:
Since there are only `n` partitions in a series file, does it really matter if `runtime.GOMAXPROCS(0) > SeriesFilePartitionN`? I don't think that will result in a concurrency > `SeriesFilePartitionN`